vyper/compiler/output_bundle.py

```diff
@@ -0,0 +1,247 @@
+import importlib
+import io
+import json
+import os
+import zipfile
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import PurePath
+from typing import Optional
+
+from vyper.compiler.input_bundle import CompilerInput, _NotFound
+from vyper.compiler.phases import CompilerData
+from vyper.compiler.settings import Settings
+from vyper.exceptions import CompilerPanic
+from vyper.semantics.analysis.module import _is_builtin
+from vyper.utils import get_long_version
+
+# data structures and routines for constructing "output bundles",
+# basically reproducible builds of a vyper contract, with varying
+# formats. note this is similar but not exactly analogous to
+# `input_bundle.py` -- the output bundle defined here contains more
+# information.
+
+
+def _anonymize(p: str):
+    segments = []
+    # replace ../../../a/b with 0/1/2/a/b
+    for i, s in enumerate(PurePath(p).parts):
+        if s == "..":
+            segments.append(str(i))
+        else:
+            segments.append(s)
+    return str(PurePath(*segments))
+
+
+# data structure containing things that should be in an output bundle,
+# which is some container containing the information required to
+# reproduce a build
+@dataclass
+class OutputBundle:
+    def __init__(self, compiler_data: CompilerData):
+        self.compiler_data = compiler_data
+
+    @cached_property
+    def compilation_target(self):
+        return self.compiler_data.compilation_target._metadata["type"]
+
+    @cached_property
+    def _imports(self):
+        return self.compilation_target.reachable_imports
+
+    @cached_property
+    def compiler_inputs(self) -> dict[str, CompilerInput]:
+        inputs: list[CompilerInput] = [
+            t.compiler_input for t in self._imports if not _is_builtin(t.qualified_module_name)
+        ]
+        inputs.append(self.compiler_data.file_input)
+
+        sources = {}
+        for c in inputs:
+            path = os.path.relpath(str(c.resolved_path))
+            # note: there should be a 1:1 correspondence between
+            # resolved_path and source_id, but for clarity use resolved_path
+            # since it corresponds more directly to zipfile semantics.
```

Comment:
why is there a mention of zipfile semantics when `OutpuBundle` should be generalized?

Reply:
old comment